source/features/open-all-notifications.js

```diff
@@ -1,60 +1,106 @@
 
-	if (unreadCount === 0) {
-		return;
-	}
+	faceboxTrigger.addEventListener('click', async () => {
+		await safeElementReady('#facebox-description');
+
+		select('#facebox-description').textContent = `Are you sure you want to open ${notificationsToOpen.length} tabs?`;
+	});
+
+	document.body.append(
+		<div id="open_all_in_tabs" style={{display: 'none'}}>
+			{faceboxTrigger}
 
+			<h2 class="facebox-header" data-facebox-id="facebox-header">Are you sure?</h2>
+
+			<p data-facebox-id="facebox-description"></p>
+
+			<div class="full-button">
+				<button class="btn btn-block" id="open-all-notifications">Open all notifications</button>
+			</div>
+		</div>
+	);
+
+	delegate('#open-all-notifications', 'click', () => {
+		openNotifications(notificationsToOpen);
+		select('.js-facebox-close').click(); // Close modal
+	});
+
+	// Creating the open button on the top
 	const openButton = <a href="#open_all_in_tabs" class="btn btn-sm">Open all unread in tabs</a>;
 
+	openButton.addEventListener('click', () => {
+		tryOpeningNotifications();
+	});
+
 	// Make a button group
 	const group = select('.tabnav .float-right');
 	group.prepend(openButton);
 	groupButtons([...group.children]);
 
-	// Move out the extra node that messes with .BtnGroup-item:last-child
-	document.body.append(select('#mark_as_read_confirm_box') || '');
+	const notificationList = select('.notifications-list');
 
-	if (unreadCount < 10) {
-		openButton.addEventListener('click', openNotifications);
-	} else {
-		// Add confirmation modal
-		openButton.setAttribute('rel', 'facebox');
-		document.body.append(
-			<div id="open_all_in_tabs" style={{display: 'none'}}>
-				<h2 class="facebox-header" data-facebox-id="facebox-header">Are you sure?</h2>
+	// Creating the open button for each repo
+	const repoNotificationContainers = select.all('.boxed-group', notificationList);
 
-				<p data-facebox-id="facebox-description">Are you sure you want to open {unreadCount} tabs?</p>
+	repoNotificationContainers.forEach(repoNotificationContainer => {
+		const actions = select('.boxed-group-action', repoNotificationContainer);
+		const firstActionButton = select('button', actions);
 
-				<div class="full-button">
-					<button class="btn btn-block" id="open-all-notifications">Open all notifications</button>
-				</div>
-			</div>
+		const repo = select('.notifications-repo-link', repoNotificationContainer).textContent.split('/')[1];
+
+		const openNotificationsButton = (
+			<button type="button" class="js-open-all-notifications" aria-label={`Open all ${repo} notifications in tabs`}>
+				{icons.externalLink()}
+			</button>
 		);
 
-		delegate('#open-all-notifications', 'click', () => {
-			openNotifications();
-			select('.js-facebox-close').click(); // Close modal
+		openNotificationsButton.addEventListener('click', () => {
+			tryOpeningNotifications(repoNotificationContainer);
 		});
-	}
+
+		actions.insertBefore(openNotificationsButton, firstActionButton);
```

Comment:
`actions.prepend` cannot be used because there is already notification paragraph before the button

![image](https://user-images.githubusercontent.com/12368291/35924390-f408213a-0c48-11e8-83e6-b8612f044459.png)